spec/models/manageiq/providers/hawkular/middleware_manager/event_catcher/runner_spec.rb

```diff
@@ -0,0 +1,83 @@
+describe ManageIQ::Providers::Hawkular::MiddlewareManager::EventCatcher::Runner do
+  subject do
+    _guid, _server, zone = EvmSpecHelper.create_guid_miq_server_zone
+    auth                 = AuthToken.new(:name     => "test",
+                                         :auth_key => "valid-token",
+                                         :userid   => "jdoe",
+                                         :password => "password")
+    ems                  = FactoryGirl.create(:ems_hawkular,
+                                              :hostname        => 'localhost',
+                                              :port            => 8080,
+                                              :authentications => [auth],
+                                              :zone            => zone)
+    described_class.new(:ems_id => ems.id)
+  end
+
+  before do
+    allow_any_instance_of(ManageIQ::Providers::Hawkular::MiddlewareManager)
+      .to receive_messages(:authentication_check => [true, ""])
+    allow_any_instance_of(MiqWorker::Runner).to receive(:worker_initialization)
+  end
+
+  context "whitelist" do
+    require 'hawkular_all'
+
+    it "is on the whitelist" do
+      event      = ::Hawkular::Alerts::Event.new({})
+      event.tags = {'miq.event_type' => 'hawkular_event.critical'}
+      expect(subject.send(:whitelist?, event)).to be true
+    end
+
+    it "is not on the whitelist" do
```

Comment:
The idea is to also serve also as documentation. I always try to think of running this command
```bash
❯ rspec --format=documentation spec/models/manageiq/providers/hawkular/middleware_manager/event_catcher/runner_spec.rb

ManageIQ::Providers::Hawkular::MiddlewareManager::EventCatcher::Runner
  whitelist
    has no tags
    is on the whitelist
    is not on the whitelist
  event_hash
    has required fields
    has optional fields
```

then the nested `it` should describe a feature of that method. 
In this case, how about `it "rejects event with unknown tag"`

the same applies for the other context/it in this spec